src/main/java/com/aws/greengrass/builtin/services/mqttproxy/MqttProxyIPCAgent.java

```
@@ -187,90 +219,154 @@ public SubscribeToIoTCoreResponse handleRequest(SubscribeToIoTCoreRequest reques
+                                    .exceptionally(
+                                            (t) -> {
+                                                LOGGER.atError()
+                                                        .cause(t)
+                                                        .kv(TOPIC_KEY, topic)
+                                                        .kv(COMPONENT_NAME, serviceName)
+                                                        .log("Unable to subscribe to topic");
+                                                throw new ServiceError(
+                                                        String.format(
+                                                                "Subscribe to topic %s failed with error %s",
+                                                                topic, t));
+                                            })
+                                    .thenApply(
+                                            (i) -> {
+                                                if (i != null && !i.isSuccessful()) {
+                                                    String rcString =
+                                                            SubAckPacket.SubAckReasonCode
+                                                                    .UNSPECIFIED_ERROR
+                                                                    .name();
+                                                    try {
+                                                        rcString =
+                                                                SubAckPacket.SubAckReasonCode
+                                                                        .getEnumValueFromInteger(
+                                                                                i.getReasonCode())
+                                                                        .name();
+                                                    } catch (RuntimeException ignored) {
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Swallowing exceptions without logging can make it difficult to identify and troubleshoot issues, potentially leading to application instability, hidden bugs, or unexpected behavior. To avoid this, ensure that all exceptions are logged using appropriate logging levels. When logging exceptions, include the exception object or its stack trace to capture detailed information. For critical exceptions, rethrow them when necessary to provide sufficient context for effective troubleshooting.

Suggested remediation:
Handle all exceptions by creating a log for each exception using the message thrown.
```
@@ -279,2 +279,3 @@
                                                     } catch (RuntimeException ignored) {
+                                                        log.error(&#34;RuntimeException is caught.&#34;, ignored);
                                                     }

```